src/Input/DownshiftSelect/Select.tsx

```diff
@@ -0,0 +1,355 @@
+    onFocusInternal();
+  };
+  const onBlur = () => {
+    if (isFunction(onBlurExternal)) {
+      onBlurExternal();
+    }
+    onBlurInternal();
+  };
+
+  return (
+    <Container data-testid="container">
+      {label && (
+        <Label
+          {...getLabelProps()}
+          data-active={isFocused}
+          data-disabled={disabled}
+          data-invalid={invalid}
+          data-testid="label"
+        >
+          {label}
+        </Label>
+      )}
+      <Combobox
+        {...getComboboxProps()}
+        data-disabled={disabled}
+        data-invalid={invalid}
+        data-active={isFocused}
+        onClick={disabled ? noop : openMenu}
+        data-testid="combobox"
+      >
+        <Input
+          {...getInputProps({
+            disabled: disabled,
+            readOnly: disableTyping,
+            placeholder: placeholder,
+            onFocus,
+            onBlur,
+            ...props,
+          })}
+          aria-invalid={invalid}
+          {...(helperText && { 'aria-describedby': helperId })}
+          {...(selectedItem && { title: selectedItem.label })}
+          data-testid="input"
+        />
+        <IndicatorsContainer data-testid="indicators-container">
+          {isLoading && <LoadingIndicator data-testid="loading-indicator" />}
+          {selectedItem && (
+            <ClearButton
+              type="button"
+              onClick={handleClear}
+              aria-label="clear selection"
+              data-testid="clear-button"
+            >
+              <CloseCircleSolidIcon />
+            </ClearButton>
+          )}
+          <ToggleButton
+            type="button"
+            {...getToggleButtonProps({
+              disabled,
+              onClick: e => e.stopPropagation(),
+            })}
+            aria-label="toggle menu"
+            data-testid="toggle-button"
+            // getToggleButtonProps adds a ref here, but functional components don't support them
+            ref={undefined}
+          >
+            {isOpen ? <ArrowUpIcon /> : <ArrowDownIcon />}
+          </ToggleButton>
+        </IndicatorsContainer>
+      </Combobox>
+      <Menu {...getMenuProps()} data-testid="menu">
+        {isOpen && displayItems.length > 0 ? (
+          displayItems.map((item, index) => (
+            <Item
+              {...getItemProps({
+                item,
+                index,
+                disabled: disabled || item.disabled,
+              })}
+              key={item.value}
+              item={item}
+              title={item.label}
+              // getToggleButtonProps adds a ref here, but functional components don't support them
+              ref={undefined}
+            >
+              {itemToString(item)}
+            </Item>
+          ))
+        ) : (
+          <EmptyList data-testid="empty-list">{emptyListText}</EmptyList>
+        )}
+      </Menu>
+      {helperText && (
+        <HelperText
+          id={helperId}
+          data-invalid={invalid}
```

Comment:
Sorry don't know what this data-invalid means?

Reply:
Oh is it for the styling?

Reply:
Yes just for the styling. The `data-` attributes generally don't have _any_ meaning unless we give them one.

Reply:
Oh got it!